tests/lib/tools/system-state

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: is-core"
+    echo "       is-core-16"
+    echo "       is-core-18"
+    echo "       is-core-20"
+    echo "       is-classic"
+    echo "       is-trusty"
+    echo "       is-xenial"
+    echo "       is-bionic"
+    echo "       is-focal"
+    echo ""
+    echo "Get general information about the current system"
+}
+
+is_core(){
+    grep "NAME=\"Ubuntu Core\"" /etc/os-release
+}
+
+is_core16(){
+    grep "PRETTY_NAME=\"Ubuntu Core 16\"" /etc/os-release
+}
+
+is_core18(){
+    grep "PRETTY_NAME=\"Ubuntu Core 18\"" /etc/os-release
+}
+
+is_core20(){
+    grep "PRETTY_NAME=\"Ubuntu Core 20\"" /etc/os-release
+}
+
+is_classic(){
+    ! is_core
```

Comment:
Note, this is fine, as it is the only and last expression in the function. That works correctly, even considering all the weird things in shell.

Reply:
nice :)